### ISSUE TITLE
[skyrl-train][algorithm] Apply inference engine top-k mask to trainer

### ISSUE THREAD

**Issue by erictang000** [NovaSky-AI/SkyRL#877]
As described in the Deepseek v3.2 Tech report - if using top_p or top_k sampling, the sampling mask should be propagated to the trainer to avoid drifting off policy in the trainer compared to the inference engine.

<img width="1015" height="301" alt="Image" src="https://github.com/user-attachments/assets/6302bc48-3c54-4cf2-aecb-56eba0fff532" />

**Comment by devpatelio** [comment]
@erictang000 From my understanding, there is no actual way to extract the mask or logits directly from vLLM (can't even recompute the mask even if you wanted to) - I'll implement a patch for now but will also open an issue in vLLM so we can remove the patch in the future.

**Comment by aoshen02** [comment]
> [@erictang000](https://github.com/erictang000) From my understanding, there is no actual way to extract the mask or logits directly from vLLM (can't even recompute the mask even if you wanted to) - I'll implement a patch for now but will also open an issue in vLLM so we can remove the patch in the future.

Will look into this in vLLM, please also directly ping me in sky rl slack channel next time.

**Comment by aoshen02** [comment]
Hi, we've enabled this feature in https://github.com/vllm-project/vllm/pull/49577, feel free to try.
